src/main/java/gov/nasa/jpl/mbee/mdk/json/util/MMSJsonParserFactory.java

```diff
@@ -0,0 +1,46 @@
+package gov.nasa.jpl.mbee.mdk.json.util;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.io.IOContext;
+import com.fasterxml.jackson.databind.MappingJsonFactory;
+
+import java.io.*;
+
+public class MMSJsonParserFactory extends MappingJsonFactory {
+
+    @Override
+    public JsonParser createParser(File f) throws IOException, JsonParseException {
+        // true, since we create InputStream from File
+        IOContext ctxt = _createContext(f, true);
+        InputStream in = new FileInputStream(f);
+        MMSJsonParser parser = (MMSJsonParser) _createParser(_decorate(in, ctxt), ctxt);
+        parser.configureClose(f);
+        return parser;
+    }
+
+    @Override
+    protected JsonParser _createParser(InputStream in, IOContext ctxt) throws IOException {
```

Comment:
All of the non-(temp)file usages are out of scope for this custom factory and should be left unimplemented, so future development uses the appropriate - default - factory when appropriate by failing early when this one is used.